newbank/server/NewBank.java

```diff
@@ -235,19 +246,56 @@ private String deleteAccount(Customer customer, String request) throws Exception
 
   }
 
+  /**
+   * Wraps the pay command method for customers so that an bank employee can specify the customer in
+   * behalf of whom the transaction is made, for example to make payments that exceed the daily
+   * transfer limit.
+   *
+   * @param bankEmployee The bank employee authorised to perform the transaction
+   * @param request      The request which is processed
+   * @return Messages indicating the success or failure of the command
+   * @throws Exception
+   */
+  private String bankEmployeePayAdapter(User bankEmployee, String request) throws Exception {
+    try {
+      String[] command = request.split(" ");
+      String onBehalfOf = "";
+      if (command.length > 1) {
+        onBehalfOf = command[1];
+        User onBehalfOfUser = users.get(onBehalfOf);
+        if (onBehalfOfUser != null && onBehalfOfUser instanceof Customer) {
+          Customer onBehalfOfCustomer = (Customer) onBehalfOfUser;
+          request = request.replaceFirst(" " + onBehalfOf, "");
+          return payAmountToOtherCustomer(onBehalfOfCustomer, request, onBehalfOf, bankEmployee);
+        }
+      }
+      return String.format(
+          "Invalid command, specify the affected customer as an additional parameter directly after the PAY command.");
+    } catch (Exception e) {
+      throw new Exception("Something went wrong with a bank employee pay command.");
+    }
+  }
+
   /**
    * Pay an amount to another customer of the bank. The outgoing account should be specified and if
    * the receiving customer or account cannot be identified or if there are no sufficient funds,
    * then the request should be rejected
    *
-   * @param customer The customer that the pay amount is paid from
-   * @param request  The pay request as recorded from the CLI interface
+   * @param customer     The customer that the pay amount is paid from
+   * @param request      The pay request as recorded from the CLI interface
+   * @param userId       Id used to check for transfer limits
+   * @param bankEmployee A bank employee that may process the transfer (no transfer limits
+   *                     respected)
    * @return "SUCCESS" if the pay request has been completed successfully. An error message will be
    * returned otherwise
    */
-  private String payAmountToOtherCustomer(Customer customer, String request) throws Exception {
+  private String payAmountToOtherCustomer(Customer customer, String request, String userId,
+      User bankEmployee) throws Exception {
     String[] requestParameterArr = request.split(" ");
 
+    boolean SPECIAL_AUTHORISATION =
```

Comment:
Maybe this could be written in small letters as it is not a constant but a variable. Might also be simpler to add it as an argument to the method and then call the method with `specialAuthorisation=true` when called from the `bankEmployeePayAdapter` method and `false` when called from `processRequest`? I think this should give the same control flow?

Reply:
Done, I tried to highlight security related variables but that was obviously not a best practice.

Reply:
Yeah that is true that it might need extra caution because of the security sensitive nature